previousnext/agov
# 27776 landings v2

*Slightly different branch requires a new PR - Not currently ready to merge*

This PR adds a new Landing Page content type, which allows the creation of workflowed landing pages powered by Paragraphs and some custom entity logic. It includes Paragraph Bundles for adding Section widgets to Landing Pages, which themselves contain Section Items, and have per-entity layout. In this way, tiled sections can be created on a landing page.

This is a demonstration of the functionality. With the addition of other widget types, a significant number of landing pages could be designed.

Workflow and revisioning works right down to the individual Section Item level.

To test, try and build a section homepage:
* The modules should install by default
* Add a new Landing Page from Admin > Content > Add Content
* Add a new "Section" to the Content field
* Name the section (e.g. "In this section...")
* In that Section, add new "Section Items". About 4 is a good number to start with.
* Switch the layout to "Tiles (3 across)" or similar
* Save

You should see a Tiled, 3 across layout in desktop view, becoming responsive at lower resolutions.

Still to do:
* ~~Titles need custom heading formatters (e.g. H2, H3) and way to link to associated link field. This applies to both Section and Section Item tiles, as well as other Paragraph bundle tiles. Not complex.~~ Done
* Permissions
* Fix layout behaviour. The class is not applying due to some issue with a preprocess hook.
* ~~Additional standard Paragraph types (e.g. Slider, View, Block, etc)~~ Done
* Advanced paragraph types
* Replace existing landing pages

![2014-12-04 at 1 06 am](https://cloud.githubusercontent.com/assets/40190/5281461/c38258b8-7b51-11e4-90eb-6ee053eae1f9.png)
![2014-12-04 at 1 06 am](https://cloud.githubusercontent.com/assets/40190/5281478/ec147f2c-7b51-11e4-855d-06167b842304.png)
![2014-12-04 at 1 08 am](https://cloud.githubusercontent.com/assets/40190/5281496/262e6c4a-7b52-11e4-9178-d55e751e44c8.png)

Example of final result:
![2014-12-04 at 1 09 am](https://cloud.githubusercontent.com/assets/40190/5281502/384abb5e-7b52-11e4-840b-93606d13bdf7.png)
